compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/collectors/components/ErrorNodeDiagnosticCollectorComponent.kt

```diff
@@ -80,8 +80,9 @@ class ErrorNodeDiagnosticCollectorComponent(collector: AbstractDiagnosticCollect
         context: CheckerContext,
         qualifiedAccessSource: FirSourceElement? = null
     ) {
-        // Will be handled by [FirDestructuringDeclarationChecker]
-        if (source.elementType == KtNodeTypes.DESTRUCTURING_DECLARATION_ENTRY) {
+        if (source.elementType == KtNodeTypes.DESTRUCTURING_DECLARATION_ENTRY || // Handled by [FirDestructuringDeclarationChecker]
+            source.kind == FirFakeSourceElementKind.DelegatedPropertyAccessor // Handled by [FirDelegatedPropertyChecker]
```

Comment:
As of https://github.com/JetBrains/kotlin/pull/4286/files#diff-9177f36c606c764bbcf6b938b75cb5466eb0b221492983a216543bc9a0fcd0ae, the below check about diagnostic kinds will be gone. So, it would be better to add a separate, `FirDelegatedPropertyChecker`-dedicated check.

Reply:
Thanks for the heads up! I will be careful when resolving conflicts after your change is merged.